lnwire/payment_initiation.go

```diff
@@ -0,0 +1,104 @@
+package lnwire
+
+import (
+	"fmt"
+	"io"
+)
+
+// PaymentInitiation is the message sent by Alice to Bob when she wishes to make payment.
```

Comment:
The invoice or sender/receiver negotiation protocol should be a layer above, rather than within the network layer itself. 

Instead nodes should house an additional "payment negotiation" server which is used to communicate the details/requirements of a payment. Note that with this design, several nodes can share the same payment server as its decoupled from the network itself. So for example BitPay could host a payment server which communicates with several of its customer nodes directly, or which handles the negotiation for any "virtual channels" it has advertised. 

Also, the latest version of master includes a simple invoice workflow that's persisted to disk and cached in memory.